pandas/core/indexing.py

```diff
@@ -186,33 +188,47 @@ def __setitem__(self, key, value):
         indexer = self._get_setitem_indexer(key)
         self._setitem_with_indexer(indexer, value)
 
-    def _has_valid_type(self, k, axis):
-        raise NotImplementedError()
+    def _validate_key(self, key, axis):
+        """
+        Ensure that key is valid for current indexer.
+
+        Parameters
+        ----------
+        key : scalar, slice or list-like
+            The key requested
+
+        axis : int
+            Dimension on which the indexing is being made
+
+        Returns:
```

Comment:
I think we dont' put this if it returns None, @TomAugspurger

Reply:
Typically we don't.